src/main/java/org/truffleruby/language/arguments/keywords/KeywordDescriptor.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 2.0, or
+ * GNU General Public License version 2, or
+ * GNU Lesser General Public License version 2.1.
+ */
+package org.truffleruby.language.arguments.keywords;
+
+public interface KeywordDescriptor {
```

Comment:
Should be an abstract class, in general `instanceof Interface` or casts are pretty slow otherwise and result in PE warnings

Reply:
We specifically do not `instanceof KeywordDescriptor` anywhere. The interface is only there for documenting the relationship.

Reply:
It's used as a type in many many places, let's make it an abstract class so e.g., even the interpreter has a chance to inline some of that. It also likely makes the SVM analysis more precise early on.

Reply:
@nirvdrum